File: 백준/Silver/1913. 달팽이/달팽이.cc
#include <iostream>
using namespace std;

int N, target;
int board[1000][1000];
int dr[4] = {1, 0, -1, 0};
int dc[4] = {0, 1, 0, -1};
int tr, tc;


int main(void) {
    cin >> N >> target;
    int cur = N * N;
    int r = 0;
    int c = 0;
    int dir = 0;
    while (cur) {
        board[r][c] = cur;
        if (cur == target) {
            tr = r;
            tc = c;
        }
        cur--;
        int nr = r + dr[dir];
        int nc = c + dc[dir];
        if (nr >= 0 && nc >= 0 && nr < N && nc < N && board[nr][nc] == 0) {
            r = nr;
            c = nc;
        } else {
            dir = (dir + 1) % 4;
            r += dr[dir];
            c += dc[dir];
        }
    }
    for (int r = 0; r < N; r++) {
        for (int c = 0; c < N; c++) {
            printf("%d ", board[r][c]);
        }
        printf("\n");
    }
    printf("%d %d\n", tr + 1, tc + 1);
}

File: 백준/Silver/1913. 달팽이/달팽이.py

N = int(input())
target = int(input())
board = [[0] * N for _ in range(N)]
dirs = ((1, 0), (0, 1), (-1, 0), (0, -1))
direction = 0
value = N**2
row = 0
col = 0
target_row = 0
target_col = 0
while value:
    if value == target:
        target_row = row
        target_col = col
    board[row][col] = value
    value -= 1

    nr = row + dirs[direction][0]
    nc = col + dirs[direction][1]
    if 0 <= nr < N and 0 <= nc < N and board[nr][nc] == 0:
        row = nr
        col = nc
        continue
    else:
        direction = (direction + 1) % 4
        row += dirs[direction][0]
        col += dirs[direction][1]
for row in board:
    print(*row, sep=' ')
print(target_row + 1, target_col + 1)
